pkg/patterns/addon/application.go

```diff
@@ -42,8 +43,22 @@ const (
 
 // TransformApplicationFromStatus modifies the Application in the deployment based off the CommonStatus
 func TransformApplicationFromStatus(ctx context.Context, instance declarative.DeclarativeObject, objects *manifest.Objects) error {
-	addonObject, ok := instance.(addonsv1alpha1.CommonObject)
-	if !ok {
+	var version string
+	var healthy bool
+	var addonObject addonsv1alpha1.CommonObject
+
+	if unstruct, ok := instance.(*unstructured.Unstructured); ok {
+		v, _, err := unstructured.NestedString(unstruct.Object, "spec", "version")
+		if err != nil {
+			return fmt.Errorf("unable to get version from unstuctured: %v", err)
+		}
+		version = v
+
+		healthy, _, err = unstructured.NestedBool(unstruct.Object, "spec", "healthy")
```

Comment:
I think this should be "status" "healthy".  We probably also should check the error (as you do with version)